src/platform/utilities/sso/index.js

```diff
@@ -1,50 +1,36 @@
-import * as Sentry from '@sentry/browser';
-
 import environment from 'platform/utilities/environment';
 import localStorage from '../storage/localStorage';
 import { hasSessionSSO } from '../../user/profile/utilities';
 import mockKeepAlive from './mockKeepAliveSSO';
 import liveKeepAlive from './keepAliveSSO';
 
 const keepAlive = environment.isLocalhost() ? mockKeepAlive : liveKeepAlive;
-let ssoSessionLength = 9000; // milliseconds
+const keepAliveThreshold = 5 * 60 * 1000; // 5 minutes, in milliseconds
 
 export async function ssoKeepAliveSession() {
-  try {
-    const res = await keepAlive();
-    const hasSSOsession = res.headers.get('session-alive');
-    if (hasSSOsession === 'true') {
-      // 'session-timeout' is in seconds, convert to milliseconds
-      ssoSessionLength = res.headers.get('session-timeout') * 1000;
-
-      const expirationTime = new Date();
-      expirationTime.setTime(expirationTime.getTime() + ssoSessionLength);
-      localStorage.setItem('sessionExpirationSSO', expirationTime);
-      localStorage.setItem('hasSessionSSO', true);
-    } else {
-      localStorage.removeItem('hasSessionSSO');
-    }
-  } catch (err) {
-    Sentry.withScope(scope => {
-      scope.setExtra('error', err);
-      Sentry.captureMessage(`SSOe error: ${err.message}`);
-    });
+  const ttl = await keepAlive();
+  if (ttl > 0) {
+    // ttl is positive, user has an active session
+    const expirationTime = new Date();
+    // 'ttl' is in seconds, convert to milliseconds
+    expirationTime.setTime(expirationTime.getTime() + ttl * 1000);
```

Comment:
I think I commented this before so feel free to ignore, but I try not to do any arithmetic with plain JS Dates bc of browser quirkiness

Reply:
converted to moment after clearing up performance concerns 👍